src/custom/utils/metadata.ts

```diff
@@ -49,3 +59,17 @@ export function generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
     },
   }
 }
+
+export async function uploadMetadataDocToIpfs(appDataDoc: AppDataDoc): Promise<string> {
+  try {
+    const client = create({ url: 'https://ipfs.infura.io:5001/api/v0' })
```

Comment:
Could load that as and ENV var/constant?
Similar to how the backend endpoints are loaded https://github.com/gnosis/cowswap/blob/afb980a9bd0194789a109d22de1ef7b2eb262ea1/src/custom/api/gnosisProtocol/api.ts#L31
and app data hash 
https://github.com/gnosis/cowswap/blob/e12c1c8393005f5a6e5490291bd4dfe15bba1a9b/src/custom/constants/appDataHash.ts

I think it would make sense to have a module under `/src/custom/constants/ipfs.ts` for loading process.env and exporting it in the app.
That applies to the PINATA variables.

What do you think?